iree/compiler/Dialect/HAL/Target/LLVM/internal/RiscvLinkerTool.cpp

```diff
@@ -0,0 +1,122 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "iree/compiler/Dialect/HAL/Target/LLVM/LinkerTool.h"
+#include "llvm/ADT/Triple.h"
+#include "llvm/IR/Function.h"
+#include "llvm/IR/IRBuilder.h"
+#include "llvm/Support/FormatVariadic.h"
+#include "llvm/Support/Host.h"
+
+#define DEBUG_TYPE "llvmaot-linker"
+
+namespace mlir {
+namespace iree_compiler {
+namespace IREE {
+namespace HAL {
+
+using llvm::Triple;
+
+// RISCV linker (ld-like); for ELF files.
+class RiscvLinkerTool : public LinkerTool {
+ public:
+  using LinkerTool::LinkerTool;
+
+  std::string getToolPath() const override {
+    // First check for setting the linker explicitly. For RISC-V targets, it has
+    // to be specified.
+    auto toolPath = LinkerTool::getToolPath();
+    if (!toolPath.empty()) return toolPath;
+
+    llvm::errs() << "IREE_LLVMAOT_LINKER_PATH environment variable must be set"
```

Comment:
This is just enforcing setting the env linker path, You can follow Android linker https://github.com/google/iree/blob/main/iree/compiler/Dialect/HAL/Target/LLVM/internal/AndroidLinkerTool.cpp#L84
and query one of the environment variables you use to build the runtime for RISK-V. In the android linker for example we set `ANDRID_NDK` to build the android runtime then the same variable is used again in linker.

Reply:
Thanks for the suggestion. For RISC-V the toolchain was defined in a custom cmake file so originally there's no environment variable to be set. I followed Android's example and set one for RISC-V. Also update the doc and the bootstrap script to reflect that. Please take a look.

Reply:
Thanks @hcindyl